plugins/modules/system/ssh_config.py

```diff
@@ -0,0 +1,289 @@
+                self.identity_file = os.path.join(dirname, self.identity_file)
+
+            if not os.path.exists(self.identity_file) and not os.path.exists(os.path.expanduser(self.identity_file)):
+                self.module.fail_json(msg='IdentityFile %s does not exist' % self.params['identity_file'])
+
+    def ensure_state(self):
+        hosts_result = self.config.search_host(self.host)
+        state = self.params['state']
+        args = dict(
+            hostname=self.params.get('hostname'),
+            port=self.params.get('port'),
+            identity_file=self.params.get('identity_file'),
+            user=self.params.get('remote_user'),
+            strict_host_key_checking=self.params.get('strict_host_key_checking'),
+            user_known_hosts_file=self.params.get('user_known_hosts_file'),
+            proxycommand=self.params.get('proxycommand'),
+        )
+
+        config_changed = False
+        hosts_changed = []
+        hosts_removed = []
+        hosts_added = []
+
+        if hosts_result:
+            for host in hosts_result:
+                if state == 'absent':
+                    # Delete host from the configuration
+                    config_changed = True
+                    hosts_removed.append(host['host'])
+                    self.config.delete_host(host['host'])
+                else:
+                    # Update host in the configuration
+                    changed, options = self.change_host(host['options'], **args)
+
+                    if changed:
+                        config_changed = True
+                        self.config.update_host(host['host'], options)
+                        hosts_changed.append({
+                            host['host']: {
+                                'old': host['options'],
+                                'new': options,
+                            }
+                        })
+        elif state == 'present':
+            changed, options = self.change_host(dict(), **args)
+
+            if changed:
+                config_changed = True
+                hosts_added.append(self.host)
+                self.config.add_host(self.host, options)
+
+        if config_changed and not self.module.check_mode:
+            try:
+                self.config.write_to_ssh_config()
+            except PermissionError as perm_exec:
+                self.module.fail_json(msg="Failed to write to %s due to permission issue: %s" % (self.config_file, to_native(perm_exec)))
+            # Make sure the file is owned by the right user and group
+            self.module.set_owner_if_different(self.config_file, self.user, False)
+            self.module.set_group_if_different(self.config_file, self.group, False)
+            perm_mode = '0600'
+            if self.config_file == '/etc/ssh/ssh_config':
+                perm_mode = '0644'
+            self.module.set_mode_if_different(self.config_file, perm_mode, False)
```

Comment:
How about first setting permissions, and then user/group? That would be a bit safer (since potential other access would be removed first).